src/transport/raw/MessageHeader.cpp

```diff
@@ -266,9 +269,11 @@ CHIP_ERROR PayloadHeader::Encode(uint8_t * data, size_t size, uint16_t * encode_
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
     uint8_t * p    = data;
+    uint8_t header = mExchangeFlags.Raw();
```

Comment:
I would drop the extra stack variable and do `Write8(p, mExchangeFlags.Raw())`. Saves a level of indirection.